src/main/java/org/jitsi/videobridge/cc/BitrateController.java

```diff
@@ -599,20 +587,29 @@ public void constraintsChanged()
      * {@link ConferenceSpeechActivity}.
      * @param bweBps the current bandwidth estimation (in bps)
      */
-    private synchronized void update(List<AbstractEndpoint> conferenceEndpoints, long bweBps)
+    private synchronized void update()
     {
-        lastEndpointOrdering = conferenceEndpoints;
-        lastBwe = bweBps;
+        long nowMs = System.currentTimeMillis();
+
+        long bweBps = getAvailableBandwidth(nowMs);
 
         logger.debug("TEMP: BitrateController " + hashCode() + " updating. " +
                 "bwebps is " + bweBps);
 
         // Create a copy as we may modify the list in the prioritize method.
-        conferenceEndpoints = new ArrayList<>(conferenceEndpoints);
+        List<AbstractEndpoint> sortedEndpoints = new ArrayList<>(sortedEndpointIds.size());
+        for (String endpointId : sortedEndpointIds)
+        {
+            AbstractEndpoint abstractEndpoint = destinationEndpoint.getConference().getEndpoint(endpointId);
```

Comment:
why did we change the member to hold string endpoint ids instead of endpoints if this method is just going to get the endpoints?

Reply:
can't holding the endpoints lead to memory leaks?

Reply:
so maybe weakreference is the right way to go?